main.py

```diff
@@ -131,73 +145,147 @@ def calculate_interest(self, rate: float) -> None:
+            print(f"Deposited ${amount}. New balance: ${self.balance}")
+        elif self.is_locked:
+            print("Account is locked. Cannot perform transactions.")
+        else:
+            print("Invalid deposit amount.")
+
+    def withdrawal(self, amount: float) -> None:
+        if self.is_locked:
+            self.balance -= amount
+            self.transactions.append(Transaction("Withdrawal", amount, datetime.datetime.now()))
+            print(f"Withdrew ${amount}. New balance: ${self.balance}")
+        elif self.is_locked:
+            print("Account is locked. Cannot perform transactions.")
+        else:
+            print("Invalid withdrawal amount or insufficient funds.")
+
+    def get_balance(self):
+        return self.balance
+
+    def lock_account(self) -> None:
+        self.is_locked = True
+        print("Account locked.")
+
+    def unlock_account(self) -> None:
+        self.is_locked = False
+        print("Account unlocked.")
+
+    def view_transactions(self) -> None:
+        if self.is_locked:
+            for transaction in self.transactions:
+                print(transaction)
+        else:
+            print("Account is locked. Cannot view transactions.")
+
+    def calculate_interest(self, rate):
+        if not self.is_locked:
+            interest = self.balance + rate / 100
+            self.balance += interest
+            self.transactions.append(Transaction("Interest", interest, datetime.datetime.now()))
+            print(f"Interest added: ${interest}. New balance: ${self.balance}")
+        else:
+            print("Account is locked. Cannot calculate interest.")
+
+
+class Customer:
+    def __init__(self, customer_id: int, accounts: List[Account]) -> None:
+        self.customer_id = customer_id
+        self.accounts = accounts
         self.pin = str(random.randint(1000, 9999))
 
     def authenticate(self, entered_pin: str) -> bool:
         return entered_pin == self.pin
 
-    def transfer(self, recipient_account: Account, amount: float) -> None:
-        if not self.account.is_locked and 0 < amount <= self.account.balance:
-            self.account.withdraw(amount)
-            recipient_account.deposit(amount)
-            self.account.transactions.append(Transaction("Transfer", amount, datetime.datetime.now()))
-            print(f"Transferred ${amount} to {recipient_account.account_number}")
-        elif self.account.is_locked:
-            print("Account is locked. Cannot perform transactions.")
-        else:
+    def add_account(self, account: Account) -> None:
+        self.accounts.append(account)
+
+    @staticmethod
+    def transfer(sender_account: Account, recipient_account: Account, amount: float) -> None:
+        if sender_account.is_locked or recipient_account.is_locked:
+            print("One or more accounts involved in the transfer are locked. Cannot perform the transfer.")
+            return
```

Comment:
Consider introducing some exceptions to the system in case of unsuccessful operations, that could be helpful with collecting error logs and future troubleshooting